controllers/openshift_controller.go

```diff
@@ -943,31 +943,52 @@ func (r *KataConfigOpenShiftReconciler) processKataConfigDeleteRequest() (ctrl.R
 		// of operator functionality
 		_ = r.disablePeerPods()
 
+		// Handle podvm image deletion
+		// Since we want to declaratively reach the final state, we need to reconcile when there are errors
+		// as we want the system to give a chance of fixing the error.
+		// For cases we don't want to reconcile, ie for ImageDeletedSuccessfully and UnsupportedPodVMImageProvider
+		// we should just log the message and let the code continue without explicitly returning from the method
+
+		// Following are returned statuses:
+		// ImageDeletedSuccessfully
+		// UnsupportedPodVMImageProvider
+		// ImageDeletionFailed
+		// RequeueNeeded
+		// ImageDeletionStatusUnknown
+
 		// Handle podvm image deletion
 		status, err := ImageDelete(r.Client)
-		if status == RequeueNeeded && err == nil {
-			// Set the KataConfig status to PodVM Image Deleting
+		switch status {
+		case ImageDeletedSuccessfully:
+			r.setInProgressConditionToPodVMImageDeleted()
+			igLogger.Info("PodVM Image deleted successfully")
```

Comment:
Even if this is about image generation, it is the core controller code. I'd rather keep `igLogger` for the image generator code and do :

```suggestion
			r.Log.Info("PodVM Image deleted successfully")
```
as everywhere else in this file.

Reply:
good catch.. it makes sense